tests/test_transformations/test_run_umap.py

```diff
@@ -0,0 +1,68 @@
+import unittest
+import anndata
+import numpy as np
+from spac.transformations import run_umap
+
+
+class TestRunUMAP(unittest.TestCase):
+
+    def setUp(self):
+        self.adata = anndata.AnnData(
+            X=np.random.rand(10, 10).astype(np.float32)
+        )
+        self.layer = 'layer1'
+        self.adata.layers[self.layer] = (
+            np.random.rand(10, 10).astype(np.float32)
+        )
+
+    def test_typical_case(self):
+        try:
+            run_umap(self.adata, n_neighbors=5, min_dist=0.1, n_components=2,
+                     random_state=42, layer=self.layer)
+        except Exception as e:
```

Comment:
@fangliu117, why we have these exceptions in the unit tests?

Reply:
@georgezakinih Thanks for catching this issue. The try-except block in the test_typical_case method was used to capture any exceptions from run_umap and provide a custom failure message using self.fail(), which indicates that the run_umap function raised an exception during its execution. But now I realized it's redundant of catching and re-raising as a test failure here, so I simply this test into:

def test_typical_case(self):
    run_umap(self.adata, n_neighbors=5, min_dist=0.1, n_components=2, 
             random_state=42, layer=self.layer)
    self.assertIn("X_umap", self.adata.obsm)